packages/@o3r/schematics/src/rule-factories/ng-add/index.ts

```diff
@@ -1,16 +1,53 @@
 import { SchematicContext, Tree } from '@angular-devkit/schematics';
 import { NgAddPackageOptions, NodePackageNgAddTask } from '../../tasks/index';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import { PackageJson } from 'type-fest';
 
 /**
- * Ng add the packages
+ * Install via `ng add` a list of npm packages. If the list is provided via `packages` param, that list of packages will be installed.
+ * If the list is empty, the `@o3r`peer dependencies of parent package will be installed.
+ * If nor the list or the parent package are provided, nothing will be done
  *
- * @param packages
- * @param tree
- * @param context
+ * @param packages List of packages to be installed via `ng add`
+ * @param parentPackage The parent package on which the o3r peer dependencies are installed
+ * @param options install options
  */
-export function ngAddPackages(packages: string[], options?: NgAddPackageOptions) {
+export function ngAddPackages(packages: string[] = [], parentPackage?: string, options?: NgAddPackageOptions) {
   return (tree: Tree, context: SchematicContext): Tree => {
-    packages.forEach((packageName) => context.addTask(new NodePackageNgAddTask(packageName, options)));
+    let packagesToInstall = packages;
+    if (parentPackage && packages.length === 0) {
```

Comment:
I think you can do simpler than this block:

```typescript
export function ngAddPackages(packages: string[], options?: NgAddPackageOptions) {
  return (tree: Tree, context: SchematicContext): Tree => {
    if (packages.length > 0) {
      context.logger.info(`'${parentPackage || ''}' - 'ng add' has been launched for the following packages:`);
      packages.forEach((packageName) => {
        context.logger.info(`Running ng add for: ${packageName}`);
        return context.addTask(new NodePackageNgAddTask(packageName, options));
      });
    }

    return tree;
  };
}

export function getPeerDepWithPattern(packageJsonPath: string, pattern: RegExp = /^@(otter|o3r|ama-sdk)/) {
  const pck: packageJsonContent = JSON.parse(fs.readFileSync(packageJsonPath, { encoding: 'utf-8' }));
  const optionalPackages = Object.entries(packageJsonContent.peerDependenciesMeta || {})
          .filter(([, dep]) => dep?.optional)
          .map(([depName]) => depName);

  const matchingPackages = Object.keys(packageJsonContent.peerDependencies || [])
          .filter(peerDep => pattern.test(peerDep) && !optionalPackages.includes(peerDep));

  return matchingPackages;
}
```

then in your schematics you can use it with

```typescript
const deps = getPeerDepWithPattern(path.resolve(__dirname, '..', '..', 'package.json'));
ngAddPackages([...deps, /* additional if needed */], {...});
```